pkg/envtest/printer/prow.go

```diff
@@ -35,75 +32,22 @@ var (
 	allRegisteredSuitesLock = &sync.Mutex{}
 )
 
-type prowReporter struct {
-	junitReporter *reporters.JUnitReporter
-}
-
-// NewProwReporter returns a prowReporter that will write out junit if running in Prow and do
-// nothing otherwise.
-// WARNING: It seems this does not always properly fail the test runs when there are failures,
-// see https://github.com/onsi/ginkgo/issues/706
-// When using this you must make sure to grep for failures in your junit xmls and fail the run
-// if there are any.
-func NewProwReporter(suiteName string) ginkgo.Reporter {
+// AddReport adds.
+func AddReport(report ginkgo.Report, suiteName string) {
 	allRegisteredSuitesLock.Lock()
 	if allRegisteredSuites.Has(suiteName) {
 		panic(fmt.Sprintf("Suite named %q registered more than once", suiteName))
 	}
 	allRegisteredSuites.Insert(suiteName)
 	allRegisteredSuitesLock.Unlock()
 
-	if os.Getenv("CI") == "" {
-		return &prowReporter{}
-	}
 	artifactsDir := os.Getenv("ARTIFACTS")
-	if artifactsDir == "" {
-		return &prowReporter{}
-	}
-
-	path := filepath.Join(artifactsDir, fmt.Sprintf("junit_%s_%d.xml", suiteName, config.GinkgoConfig.ParallelNode))
-	return &prowReporter{
-		junitReporter: reporters.NewJUnitReporter(path),
-	}
-}
-
-func (pr *prowReporter) SpecSuiteWillBegin(config config.GinkgoConfigType, summary *types.SuiteSummary) {
-	if pr.junitReporter != nil {
-		pr.junitReporter.SpecSuiteWillBegin(config, summary)
-	}
-}
-
-// BeforeSuiteDidRun implements ginkgo.Reporter.
-func (pr *prowReporter) BeforeSuiteDidRun(setupSummary *types.SetupSummary) {
-	if pr.junitReporter != nil {
-		pr.junitReporter.BeforeSuiteDidRun(setupSummary)
-	}
-}
-
-// AfterSuiteDidRun implements ginkgo.Reporter.
-func (pr *prowReporter) AfterSuiteDidRun(setupSummary *types.SetupSummary) {
-	if pr.junitReporter != nil {
-		pr.junitReporter.AfterSuiteDidRun(setupSummary)
-	}
-}
-
-// SpecWillRun implements ginkgo.Reporter.
-func (pr *prowReporter) SpecWillRun(specSummary *types.SpecSummary) {
-	if pr.junitReporter != nil {
-		pr.junitReporter.SpecWillRun(specSummary)
-	}
-}
-
-// SpecDidComplete implements ginkgo.Reporter.
-func (pr *prowReporter) SpecDidComplete(specSummary *types.SpecSummary) {
-	if pr.junitReporter != nil {
-		pr.junitReporter.SpecDidComplete(specSummary)
-	}
-}
 
-// SpecSuiteDidEnd Prints a newline between "35 Passed | 0 Failed | 0 Pending | 0 Skipped" and "--- PASS:".
-func (pr *prowReporter) SpecSuiteDidEnd(summary *types.SuiteSummary) {
-	if pr.junitReporter != nil {
-		pr.junitReporter.SpecSuiteDidEnd(summary)
+	if os.Getenv("CI") != "" && artifactsDir != "" {
```

Comment:
The only reason for this existing btw was that generating junit made ginkgo always exist with RC 0 before, even if there were failures. If there is an easier way to get it to both generate junit and exit with rc != 0 if there is a failure, that would be great.

Reply:
I think we can run tests with the `ginkgo` cli instead of `go test`, which allows to configure junit report generation. Still evaluating if it works for everything.
It does run regular Go tests, but it does not include them in the junit report. But we don't have them in the report currently anyway, correct?

Reply:
I've switched the PR to the ginkgo cli now to see if it works.
One major drawback I've seen is that `ginkgo` doesn't cache tests, so it runs the full suite every time. That might not be ideal and break/slow down workflows.